pandas/_libs/tslibs/period.pyx

```diff
@@ -1685,7 +1685,9 @@ cdef class _Period(object):
                 if other.freq != self.freq:
                     msg = _DIFFERENT_FREQ.format(self.freqstr, other.freqstr)
                     raise IncompatibleFrequency(msg)
-                return (self.ordinal - other.ordinal) * self.freq
+                base_freq = type(self.freq)(normalize=self.freq.normalize,
+                                            **self.freq.kwds)
```

Comment:
pass `n=1` explicitly here.

add a reference `# GH#23915` for future readers

Reply:
done